src/dimred/tsne/config.vsh.yaml

```diff
@@ -0,0 +1,116 @@
+          type: string
+          description: The `.obsm` slot to use as input for the tSNE computation.
+          required: true
+          example: "X_pca"
+    
+    - name: Outputs
+      arguments:
+        - name: "--output"
+          alternatives: ["-o"]
+          type: file
+          description: Output h5mu file.
+          direction: output
+          required: true
+          example: output.h5mu
+
+        - name: "--output_compression"
+          type: string
+          description: The compression format to be used on the output h5mu object.
+          choices: ["gzip", "lzf"]
+          required: false
+          example: "gzip"
+
+        - name: "--obsm_output"
+          type: string
+          description: The pre/suffix under which to store the tSNE results.
+          default: "X_tsne"
+    
+    - name: Arguments
+      arguments:
+
+        - name: "--n_pcs"
+          type: integer
+          description: The number of principal components to use for the tSNE computation.
+          default: 50
+          required: false
+
+        - name: "--perplexity"
+          type: double
+          description: The perplexity is related to the number of nearest neighbors that is used in other manifold learning algorithms. Larger datasets usually require a larger perplexity. Consider selecting a value between 5 and 50. Different values can result in significantly different results.
+          default: 30.0
+
+        - name: "--min_dist"
+          type: double
+          description: The effective minimum distance between embedded points. Smaller values will result in a more clustered/clumped embedding where nearby points on the manifold are drawn closer together, while larger values will result on a more even dispersal of points. The value should be set relative to the spread value, which determines the scale at which embedded points will be spread out. 
+          default: 0.5
+
+        - name: "--metric"
+          type: string
+          description: Distance metric to calculate neighbors on.
+          default: euclidean
+
+        - name: "--early_exaggeration"
+          type: double
+          description: Controls how tight natural clusters in the original space are in the embedded space and how much space will be between them. For larger values, the space between natural clusters will be larger in the embedded space. Again, the choice of this parameter is not very critical. If the cost function increases during initial optimization, the early exaggeration factor or the learning rate might be too high.
+          default: 12.0
+
+        - name: "--learning_rate"
+          type: double
+          description: The learning rate for t-SNE optimization. Typical values range between 10.0 and 1000.0.
+          default: 1000.0
+
+        - name: "--random_state"
+          type: integer
+          description: The random seed to use for the tSNE computation.
+          default: 0
+
+        - name: "--n_jobs"
```

Comment:
I would remove this parameters.
In a component we have `meta['cpus']` available (similar to par[...]), that is set by viash (which also provides the argument on CLI and uses the correct value if the component is build as a nextflow module)

Reply:
So then just pass `meta['cpus']` to tsne like `sc.tl.tsne(n_jobs=meta['cpus'])`?

Reply:
Yes! 👍